src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -544,11 +544,19 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				return false;
 			}
 
+			// only keep track of input if prompt hasn't already been shown
+			if (this._storageService.getBoolean(SHOW_TERMINAL_CONFIG_PROMPT, StorageScope.GLOBAL, true) &&
+				// just a normal key with no modifier
+				event.key.match(/^[a-z0-9]$/i) && !event.ctrlKey && !event.shiftKey && !event.altKey) {
+				this._storageService.store(HAS_TYPED_IN_TERMINAL + this.id, true, StorageScope.GLOBAL, StorageTarget.USER);
```

Comment:
Does the storage service need to be involved?

Reply:
probably not. I could make the TerminalInstance have an `hasHadInput` boolean